HDSAviz/data_processing.py

```diff
@@ -0,0 +1,197 @@
+    """
+    Returns a pandas dataframe with all of the input parameters
+    set in running the model for sensitivity analysis, including
+    the parameters that are not analyzed for sensitivity (end time
+    and oxygen content).
+
+    numrows = the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows)
+
+
+def get_params(numrows=None):
+    """
+    Returns a pandas dataframe with all the parameters analyzed in
+    the sensitivity analysis, but not additional parameters.
+
+    numrows = the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows,
+                     drop=['End_time', 'Oxygen'])
+
+
+def get_results(numrows=None):
+    """
+    Returns a pandas dataframe with the results of running all of the
+    simulations for the parameters sets in `input_parameters.csv`. This
+    function drops two unused functional groups from the results file.
+
+    numrows = the number of rows of the input_parameters file to read
+          (default is to read all rows).
+    """
+    return read_file('../../HDSAviz_data/results.csv',
+                     numrows=numrows,
+                     drop=['light_aromatic_C-C', 'light_aromatic_methoxyl'])
+
+
+def get_sa(path='../../HDSAviz_data/'):
+    """
+    This function reads and processes all the sensitivity analysis results
+    in a specified folder, and returns a dictionary with the corresponding
+    dataframes.  Sensitivity analysis results should be in the default SALib
+    format and must start with the letter 'a'.
+
+    Parameters:
+    -----------
+    path : String containing the relative or absolute path of the directory
+           where analysis_*.txt files are stored.  There cannot be any
+           files or folders within this directory that start with 'a'
+           except those generated by the SALib sensitivity analysis.
+
+    Returns:
+    --------
+    sens : Dictionary where keys are the names of the various output measures
+           (one output measure per analysis file in the folder specified by
+           path).  Dictionary values are a list of pandas dataframes.  If
+           only first and total order indices were calculated in the
+           sensitivity analysis then this list will contain just one pandas
+           dataframe (sens[key][0]) and the second item in the list will be
+           False.  If second order indices are present they are in a second
+           dataframe (sens[key][1]).
+    """
+    filenames = [filename for filename in os.listdir(
+                path) if filename.startswith('a')]
+    # These two functional groups are not present in the light oil fraction
+    filenames.remove('analysis_light_aromatic-C-C.txt')
```

Comment:
Can these to two filename.remove lines be replaced with something more general to remove unwanted filenames given by the user?